File: python/examples/basic.py
"""basic.py — single capture, sync. Run: python examples/basic.py"""
import sys

from webshot import WebshotClient, WebshotThrottledError, WebshotError


def main() -> int:
    with WebshotClient() as client:
        try:
            shot = client.capture("https://example.com", format="png", mode="desktop_full")
        except WebshotThrottledError as e:
            print(f"Rate limit hit — retry in {e.retry_after}s.")
            print(f"For higher limits: email {e.contact}")
            return 2
        except WebshotError as e:
            print(f"Capture failed: {e}")
            return 1

        shot.write("example.png")
        print(f"✓ saved example.png  ({len(shot.bytes)} bytes, {shot.content_type})")
        print(f"  source={shot.source}  mode={shot.mode}")
        print(f"  credits remaining: {shot.rate_limit.remaining}/{shot.rate_limit.limit}")
        return 0


if __name__ == "__main__":
    sys.exit(main())

File: php/examples/basic.php
<?php
// basic.php — single capture, save to disk. Run: php examples/basic.php
require __DIR__ . '/../vendor/autoload.php';

use Tuxxin\Webshot\Client;
use Tuxxin\Webshot\Exception\ThrottledException;
use Tuxxin\Webshot\Exception\WebshotException;

$client = new Client();

try {
    $shot = $client->capture('https://example.com', format: 'png', mode: 'desktop_full');
} catch (ThrottledException $e) {
    fwrite(STDERR, "Rate limit hit — retry in {$e->getRetryAfter()}s.\n");
    fwrite(STDERR, "For higher limits: email {$e->getContact()}\n");
    exit(2);
} catch (WebshotException $e) {
    fwrite(STDERR, "Capture failed: {$e->getMessage()}\n");
    exit(1);
}

$shot->write('example.png');
echo "✓ saved example.png  (" . strlen($shot->bytes) . " bytes, {$shot->contentType})\n";
echo "  source={$shot->source}  mode={$shot->mode}\n";
echo "  credits remaining: {$shot->rateRemaining}/{$shot->rateLimit}\n";
